### PR TITLE
Minors changes

### PR DESCRIPTION
**Pull request by twillouer**
Some minors changes to reflect more MongoDB Server

### CODE REVIEW

File: src/main/java/com/foursquare/fongo/impl/Util.java [hunk 1]
```diff
@@ -149,6 +154,33 @@ public static boolean isPositiveInt(String s) {
       return clone;
     }
 
+    if (source instanceof LazyDBList) {
```
Comment (hoffrocket) [inline src/main/java/com/foursquare/fongo/impl/Util.java:157]: why would you see a LazyDB object here?

**Comment by hoffrocket** [comment]
looks good, but I'm why would you need to handle LazyDBObject?

**Comment by twillouer** [comment]
Hi,

because I try to test Fongo with Jongo (https://github.com/bguerout/jongo) and all was in trouble (because they use LazyDBObject and LazyDBList).

I will push my tests if you want to take a look, but with this PR, more tests are green :)

**Comment by twillouer** [comment]
Try, if you want to see : https://github.com/twillouer/jongo/tree/fongo (it's very quick and dirty). Without Lazy\* the project doesn't work (it's why I need "drop" and "insert" create collection into FongoDB for example)

**Comment by hoffrocket** [comment]
there are merge conflicts, can you resubmit?

**Comment by twillouer** [comment]
done

**Pull request merged**